### PR TITLE
test: Run ESDK .NET v4.0.1 Test Vectors in CI

### PR DESCRIPTION
**Pull request by imabhichow**
*Issue #, if available:*

*Description of changes:*

- Fetch ESDK .NET v4.0.1 Test Vectors & Run .NET v4.0.1 Test Vectors against Java in CI
- Separate Vector Tests against Master Keys (runs when `-Dmasterkey=true`) and Keyrings.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

File: codebuild/ci/net-vectors-ci.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      - aws sts get-caller-identity
+
+      # Build and deploy TestVectors to maven local
+      - cd submodules/MaterialProviders/TestVectorsAwsCryptographicMaterialProviders/
+      - make build_java && make mvn_local_deploy
+      - cd $CODEBUILD_SRC_DIR
+
+      # Fetch ESDK .NET v4.0.1 Test Vectors
+      - VECTOR_ZIP=$CODEBUILD_SRC_DIR/v4-Net-4.0.1.zip
+      - VECTORS_URL=https://github.com/awslabs/aws-encryption-sdk-test-vectors/raw/master/vectors/awses-decrypt/python-2.3.0.zip
```
Comment (josecorella) [inline codebuild/ci/net-vectors-ci.yml:34]: if this is supposed to check .nety compatability, how come we are downloading the python vectors?

**Review by josecorella — CHANGES_REQUESTED** [review]

File: codebuild/ci/vectors-ci-mkp.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+      # Build and deploy TestVectors to maven local
+      - cd submodules/MaterialProviders/TestVectorsAwsCryptographicMaterialProviders/
+      - make build_java && make mvn_local_deploy
+      - cd $CODEBUILD_SRC_DIR
+  build:
+    commands:
+      # Enable masterKey to run Test Vectors against MasterKeyProvider
+      # NOTE: We need to pass the absolute path of the vectors
+      - mvn -B -ntp install -Dgpg.skip=true -Djacoco.skip=true "-Dtest=TestVectorRunner" "-Dmasterkey=true" "-DtestVectorZip=file://$CODEBUILD_SRC_DIR/src/test/resources/aws-encryption-sdk-test-vectors/vectors/awses-decrypt/python-2.3.0.zip"
```
Comment (texastony) [inline codebuild/ci/vectors-ci-mkp.yml:35]: Question: What is the relationship between the MPL's Test Vectors, and the Master Key Provider test vectors?
Or is that all the Tests for ESDK-Java depend on the MPL Test Vectors?

B/c the Master Key Provider should not need the MPL.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -58,6 +58,66 @@ jobs:
           env-vars-for-codebuild: JAVA_ENV_VERSION
         env:
           JAVA_ENV_VERSION: ${{ matrix.platform.distribution }}${{ matrix.version }}
+  vectorTestsMasterKey:
```
Comment (texastony) [inline .github/workflows/ci.yml:61]: I thought this was just loading Keys, but it is testing the legacy Master Key Provider interfaces against the test vectors, correct?
```suggestion
  vectorTestsMasterKeyProvider:
```

File: codebuild/ci/vectors-ci.yml
Comment (texastony) [inline codebuild/ci/vectors-ci.yml:1]: Blocking: What is the difference between this file and `vectors-ci-mkp.yml`?

File: codebuild/ci/net-vectors-ci.yml (hunk 1 above)
Comment (imabhichow) [reply codebuild/ci/net-vectors-ci.yml:34]: It's a nit. I forgot to remove line 34, so I will update the PR.
The test still runs on .net `VECTOR_ZIP` in line 39.

```
mvn -B -ntp install -Dgpg.skip=true -Djacoco.skip=true "-Dtest=TestVectorRunner" "-DtestVectorZip=file://$VECTOR_ZIP"
```

**Review by texastony — COMMENTED** [review]
I am having a hard relating this to the Private PR that I thought did the same thing:
https://github.com/aws/private-aws-encryption-sdk-java-staging/pull/68/files#diff-e390fc0ad5f952462bfc117b51fd807859eac77e5f85a206b90747b3218c679f

Some common lines are here, but there are significant differences.
It has been a while since I looked at this code base... 

I think we may need to go over this in more detail.

File: codebuild/ci/vectors-ci-mkp.yml (hunk 2 above)
Comment (imabhichow) [reply codebuild/ci/vectors-ci-mkp.yml:35]: ESDK-java test vectors have a dependency on MPL test vectors; even if we use master keys, we still need to compile MPL test vectors.

File: codebuild/ci/vectors-ci.yml
Comment (imabhichow) [reply codebuild/ci/vectors-ci.yml:1]: Its purpose is to distinguish between test vectors for master keys and keyrings. So we now know when there is a failure during decryption, whether with master keys or keyrings.

File: codebuild/ci/net-vectors-ci.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+version: 0.2
+
```
Comment (texastony) [inline codebuild/ci/net-vectors-ci.yml:2]: Suggestion: These builds are all so similar, I recommend explicitly documenting what they do.
```suggestion
# Runs Only the ESDK-NET v4.0.1 Decryption Vectors, testing Required EC CMM
```

File: codebuild/ci/vectors-ci-mkp.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
+version: 0.2
+
```
Comment (texastony) [inline codebuild/ci/vectors-ci-mkp.yml:2]:
```suggestion
# Runs "Golden Manifest" against legacy Master Key Providers
```

File: codebuild/ci/vectors-ci.yml
Comment (texastony) [inline codebuild/ci/vectors-ci.yml:1]: Suggestion:  Add a comment stating what this Build does:
```
# Runs "Golden Manifest" against Keyrings
```

**Review by josecorella — APPROVED** [review]

**Pull request merged**